devito/ir/support/space.py

```diff
@@ -154,32 +144,27 @@ def relaxed(self):
         return Interval(self.dim.root, self.lower, self.upper, self.stamp)
 
     @property
-    def limits(self):
+    def offsets(self):
         return (self.lower, self.upper)
 
     def intersection(self, o):
-        if self.overlap(o):
-            return Interval(self.dim, max(self.lower, o.lower), min(self.upper, o.upper),
-                            self.stamp)
+        if self.is_compatible(o):
+            svl, svu = Vector(self.lower, smart=True), Vector(self.upper, smart=True)
```

Comment:
@georgebisbas : note I'm using Vector here (understanding why could be a nice exercise). You may start from the tests in test_ir.py